core/services/relay/relay.go

```diff
@@ -88,3 +101,28 @@ func (r *relayerAdapter) HealthReport() map[string]error {
 	maps.Copy(r.RelayerExt.HealthReport(), hr)
 	return hr
 }
+
+type ErrLogFiltersNotSupported struct {
+	relayName string
+}
+
+func (e ErrLogFiltersNotSupported) Error() string {
+	return fmt.Sprintf("Log filtering is not supported by relay %s", e.relayName)
+}
+
+func (r *relayerAdapter) OnCreateJob(rargs types.RelayArgs, q pg.Queryer) (err error) {
+	relay, ok := r.Relayer.(LogPollerCapable)
+	if !ok {
+		return nil
+	}
+	return relay.RegisterLogFilters(rargs, q)
+
+}
+
+func (r *relayerAdapter) OnDeleteJob(rargs types.RelayArgs, q pg.Queryer) (err error) {
+	relay, ok := r.Relayer.(LogPollerCapable)
+	if !ok {
+		return nil
```

Comment:
Should these return errors?

Reply:
The way I was thinking of this is "if this blockchain is `LogPollerCapable` then perform log poller specific actions, if not do nothing.  It's important that the delegate-level `OnDeleteJob()` not return an error in the case where there is nothing to be done, because that would cause the job deletion to fail... making it impossible to delete jobs on blockchains without log poller.  At the relay level, we could have it return an error and then convert it into nil at the delegate level, but... "no action necessary" doesn't seem like an error, so nil makes sense to me.

Reply:
If we change the `(LogPollerCapble)` to be `(JobHooks)`, and the `UnregisterLogFilters()` to `OnDeleteJob()` here, I think nil still makes sense.  For now, most of the relays haven't implemented `OnDeleteJob()` so, we shouldn't make that an error until we decide it's mandatory for them all to implement it, right?

Reply:
Ah yeah, nil makes sense, but then is `ErrLogFiltersNotSupported` unused?

Reply:
Yes, I meant to remove that... good catch